### PR TITLE
quo2 documentation drawer component

### LINKED ISSUE #15016: Implement Quo2  component  Drawers/ Documentation drawers
Designs:
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=13833%3A272409&t=29WaMjPTYMtHqa1n-0

Component tests
Preview Screens

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes #15016 

To test the component: quo2 preview -> drawers -> documentation drawers

### Todo
- [x] Fix scroll not working on Android.
- [x] Shell theme

### Screencast iOS

https://user-images.githubusercontent.com/19279756/233043665-b9abe39c-82c9-49c1-b676-9bcb40bfbba6.mov

### Screencast Android

https://user-images.githubusercontent.com/19279756/233043693-03b20c92-7de3-4b3a-865b-081bfedbea85.mov

### Areas that may be impacted
This PR introduces a new blur background option - for the bottom sheets. Currently, we are only using this feature in the documentation drawer component(only added in quo2 preview). However, it would be wise to test other places where bottom sheets are used in our UI to ensure that this change does not impact the existing design and functionality.

status: ready

### CODE REVIEW

File: src/quo2/components/drawers/documentation_drawers/component_spec.cljs
Comment (J-Son89) [inline src/quo2/components/drawers/documentation_drawers/component_spec.cljs]: I think it would be more readable if you change to :as documentation-drawer?

File: src/quo2/components/drawers/documentation_drawers/view.cljs
Comment (J-Son89) [inline src/quo2/components/drawers/documentation_drawers/view.cljs]: we don't need `cta?` prop, you can just do `when on-press-cta ...` below

File: src/quo2/components/drawers/documentation_drawers/view.cljs
Comment (J-Son89) [inline src/quo2/components/drawers/documentation_drawers/view.cljs]: consider something simple like "button" in place of `cta`. i.e

`on-press-button`
`button-icon`

File: src/quo2/components/drawers/documentation_drawers/style.cljs
Comment (J-Son89) [inline src/quo2/components/drawers/documentation_drawers/style.cljs:2]: hmm, this is lacking the color variations? There should be light, dark and shell themes. 🤔 or you'll do that in a follow up?

File: src/status_im2/common/bottom_sheet/view.cljs
Comment (J-Son89) [inline src/status_im2/common/bottom_sheet/view.cljs]: we use TODO and it should include a github link to that issue.
Unless this is just for your pr while it's in draft ofc :)

Comment (ajayesivan) [reply src/status_im2/common/bottom_sheet/view.cljs]: It's just for me to fix before moving the PR to ready. I will use TODO for adding future tasks 👍

File: src/quo2/components/drawers/documentation_drawers/style.cljs
Comment (ajayesivan) [reply src/quo2/components/drawers/documentation_drawers/style.cljs:2]: I will add it.

File: src/quo2/components/drawers/documentation_drawers/view.cljs
Comment (ajayesivan) [reply src/quo2/components/drawers/documentation_drawers/view.cljs]: `cta?` is used to show/hide the button.

Comment (J-Son89) [reply src/quo2/components/drawers/documentation_drawers/view.cljs]: yes but do we not only need the button when there is an on-press action 🤔 
Maybe I have it wrong :)

Comment (ajayesivan) [reply src/quo2/components/drawers/documentation_drawers/view.cljs]: Yes, we can do that. In the component design it's a 'Read more' button and when clicked it will show the full content and hide the button. So I thought it's better to add a separate prop to control the visibility of the button. But we can achieve it using `on-press`, should I remove `cta?`?

Comment (J-Son89) [reply src/quo2/components/drawers/documentation_drawers/view.cljs]: Ah yea, you're right. Let's leave it as is so 👌

File: src/quo2/components/drawers/documentation_drawers/style.cljs
Comment (ajayesivan) [reply src/quo2/components/drawers/documentation_drawers/style.cljs:2]: I'm using `text`, `button` and `bottom-sheet` components so dark & light variations will be handled by those components. For `shell` variant I have updated the bottom-sheet with a `blur?` option. Please check the the `bottom-sheet/view` file. Design only has shell variant in the dark theme so I think it will only be used in dark theme. Should we confirm this with the design team?

Comment (J-Son89) [reply src/quo2/components/drawers/documentation_drawers/style.cljs:2]: Yes, that makes sense about the components.
shell theme is another variation of dark theme, let's confirm with them just to make sure it's all setup as expected 👍 .

File: src/status_im2/common/bottom_sheet/styles.cljs
Comment (J-Son89) [inline src/status_im2/common/bottom_sheet/styles.cljs]: you will probably need QA to review this pr since this adjusts the bottom sheet in app 👍

**Review by J-Son89 — APPROVED** [review]
nice work 🎉

File: src/status_im2/common/bottom_sheet/styles.cljs
Comment (ajayesivan) [reply src/status_im2/common/bottom_sheet/styles.cljs]: @J-Son89 Do I just need to add the 'request-manual-qa' label for this, or is there anything else that needs to be done?

Comment (J-Son89) [reply src/status_im2/common/bottom_sheet/styles.cljs]: Yes, once you have a dev approval (which you now do) then you can add that label . Also good to give a comment or update description of what area should be tested

**Review by yqrashawn — APPROVED** [review]

**Comment by churik** [comment]
@ajayesivan 
Thank you for PR!
I do not see that component affects the current bottom sheets, however, noticed that if SHELL=TRUE, the color gets brown:
IMAGE 2023-05-02 13:53:10

Is it something expected for now?

**Comment by ajayesivan** [comment]
@churik Background color is correct. I have fixed the text color.




Shell theme background is the same on both light & dark themes.

**Comment by pavloburykh** [comment]
@ajayesivan thanx for the fix. Ready to be merged.

File: src/status_im2/common/bottom_sheet/styles.cljs
Comment (ibrkhalil) [inline src/status_im2/common/bottom_sheet/styles.cljs:24]: This change caused bottom-sheet selected item to not show.
Fixed in https://github.com/status-im/status-mobile/pull/15677

**Pull request merged**
